From 6855cd63d9ae9ed75e3c07674323c4f04c3471bc Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Fri, 26 Jan 2018 00:03:46 +0100
Subject: [PATCH] arm64: dts: rockchip: rk3328-roc-cc: disable sd-card voltage
 select

Voltage select should set GRF_SOC_CON10 bit 1,
vendor kernel repurpose GPIO0_D1 to signal this,
RK kernel uses GRF_SOC_CON10 bit 1 to mute avcodec.
---
 arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts b/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts
index 26405429b382..ae8bca0924dd 100644
--- a/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts
@@ -75,8 +75,6 @@
 		gpios = <&gpio0 RK_PD1 GPIO_ACTIVE_HIGH>;
 		states = <1800000 0x1
 			  3300000 0x0>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&sd_pwr_1800_sel>;
 		regulator-name = "vccio_sd";
 		regulator-type = "voltage";
 		regulator-min-microvolt = <1800000>;
@@ -412,12 +410,6 @@
 		};
 	};
 
-	sd-pwerset {
-		sd_pwr_1800_sel: sd-pwr-1800-sel {
-			rockchip,pins = <0 RK_PD1 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-	};
-
 	pmic {
 		pmic_int_l: pmic-int-l {
 			rockchip,pins = <1 RK_PD0 RK_FUNC_GPIO &pcfg_pull_up>;
@@ -445,13 +437,11 @@
 	cap-mmc-highspeed;
 	cap-sd-highspeed;
 	disable-wp;
-	max-frequency = <100000000>;
+	max-frequency = <150000000>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&sdmmc0_clk &sdmmc0_cmd &sdmmc0_dectn &sdmmc0_bus4>;
-	sd-uhs-sdr104;
 	supports-sd;
 	vmmc-supply = <&vcc_sd>;
-	vqmmc-supply = <&vccio_sd>;
 	status = "okay";
 };
 

